packages/livekit-rtc/src/rpc.ts

```diff
@@ -0,0 +1,165 @@
+// SPDX-FileCopyrightText: 2024 LiveKit, Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+// TODO: This implementation is only a prototype
+//       The final version will use a protocol types where possible
+export class RpcRequest {
+  id: string;
+  method: string;
+  payload: string;
+  responseTimeoutMs: number;
+
+  constructor({
+    method,
+    payload,
+    responseTimeoutMs,
+    id = crypto.randomUUID(),
+  }: {
+    method: string;
+    payload: string;
+    responseTimeoutMs: number;
+    id?: string;
+  }) {
+    this.id = id;
+    this.method = method;
+    this.payload = payload;
+    this.responseTimeoutMs = responseTimeoutMs;
+  }
+}
+
+export class RpcAck {
+  requestId: string;
+
+  constructor({ requestId }: { requestId: string }) {
+    this.requestId = requestId;
+  }
+}
+
+export class RpcResponse {
+  requestId: string;
+  payload?: string;
+  error?: RpcError;
+
+  constructor({
+    requestId,
+    payload,
+    error,
+  }: {
+    requestId: string;
+    payload?: string;
+    error?: RpcError;
+  }) {
+    this.requestId = requestId;
+    this.payload = payload;
+    this.error = error;
+  }
+}
+
+/**
+ * Specialized error handling for RPC methods.
+ *
+ * Instances of this type, when thrown in a method handler, will have their `message`
+ * serialized and sent across the wire. The sender will receive an equivalent error on the other side.
+ *
+ * Build-in types are included but developers may use any string, with a max length of 256 bytes.
+ */
+
+export class RpcError extends Error {
+  static MAX_MESSAGE_BYTES = 256;
+  static MAX_DATA_BYTES = 15360; // 15 KB
+
+  code: number;
```

Comment:
```suggestion
  code: ErrorCodes | number;
```

Reply:
doesn't seem to work like that - i'm a bit confused by these module-enums tbh. leaving as a number, I think its fine

Reply:
ah, probably it was `RpcError.ErrorCodes | number`

Reply:
yeah it didn't like that for some reason. I'll circle back on this in the final version, thank you!

Reply:
done